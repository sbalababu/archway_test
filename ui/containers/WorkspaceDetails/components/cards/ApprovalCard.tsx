import * as React from 'react';
import { Card, Icon } from 'antd';
import { ApprovalItem } from '../../../../models/Workspace';

/* tslint:disable:no-var-requires */
const TimeAgo = require('timeago-react').default;

interface Props {
  data?: ApprovalItem;
  children?: any;
  onApprove?: React.MouseEventHandler<HTMLAnchorElement>;
}

const ApprovalCard = ({ data, children, onApprove }: Props) => {
  const loading = data && data.status && data.status.loading;
  const approvedAt = data && data.approval_time;

  return (
    <Card style={{ height: '100%' }} bordered>
      <div style={{ textAlign: 'center', fontSize: 17 }}>{children}</div>
      <div
        style={{
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          fontWeight: 300,
          padding: '32px 0 48px 0',
        }}
      >
        <Icon
          type={approvedAt ? 'safety-certificate' : 'dash'}
          theme={approvedAt ? 'twoTone' : 'outlined'}
          style={{ fontSize: 24 }}
        />
        <div style={{ marginLeft: 8, lineHeight: 1.3 }}>
          {!!approvedAt && <div>APPROVED</div>}
          {!!approvedAt && (
            <div style={{ fontSize: 12 }}>
              <TimeAgo datetime={approvedAt} />
            </div>
          )}
          {!approvedAt &&
            (loading ? (
              <Icon theme="outlined" type="loading" style={{ fontSize: 16 }} />
            ) : (
              !!onApprove && (
                <a onClick={onApprove}>
                  {' '}
                  {/* eslint-disable-line */}
                  APPROVE
                </a>
              )
            ))}
        </div>
      </div>
    </Card>
  );
};

export default ApprovalCard;
