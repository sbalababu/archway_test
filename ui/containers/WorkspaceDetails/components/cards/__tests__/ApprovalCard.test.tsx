import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ApprovalCard from '../ApprovalCard';

describe('ApprovalCard', () => {
  it('renders correctly', () => {
    const props = {
      data: {
        approver: 'tforester',
        approval_time: new Date('2019-06-20T20:52:35Z'),
      },
    };
    const wrapper = shallow(<ApprovalCard {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
