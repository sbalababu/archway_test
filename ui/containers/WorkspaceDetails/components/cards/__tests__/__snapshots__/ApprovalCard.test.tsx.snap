// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApprovalCard renders correctly 1`] = `
<Card
  bordered={true}
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    style={
      Object {
        "fontSize": 17,
        "textAlign": "center",
      }
    }
  />
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "fontWeight": 300,
        "justifyContent": "center",
        "padding": "32px 0 48px 0",
      }
    }
  >
    <Icon
      style={
        Object {
          "fontSize": 24,
        }
      }
      theme="twoTone"
      type="safety-certificate"
    />
    <div
      style={
        Object {
          "lineHeight": 1.3,
          "marginLeft": 8,
        }
      }
    >
      <div>
        APPROVED
      </div>
      <div
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        <TimeAgo
          datetime={2019-06-20T20:52:35.000Z}
          live={true}
          locale="en"
        />
      </div>
    </div>
  </div>
</Card>
`;
